### PR TITLE
s/logprob/logdensity

### PR DESCRIPTION
**Pull request by junpenglao**
Renaming logprob to logdensity for more consistent terminology.

How: search for `logprob(?!ability)` and replace with `logdensity`, excluding everything in `docs/**`, `examples/`, `tests/**` as they are mostly logprob.

# Thank you for opening a PR!

 A few important guidelines and requirements before we can merge your PR:

 - [ ] *If I add a new sampler*, there is an issue discussing it already;
 - [x] We should be able to understand what the PR does from its title only;
 - [x] There is a high-level description of the changes;
 - [x] There are links to *all* the relevant issues, discussions and PRs;
 - [x] The branch is rebased on the latest `main` commit;
 - [x] Commit messages follow these [guidelines](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html);
 - [x] The code respects the current naming conventions;
 - [x] Docstrings follow the [numpy style guide](https://numpydoc.readthedocs.io/en/latest/format.html)
 - [x] `pre-commit` is installed and configured on your machine, and you ran it before opening the PR;
 - [x] There are tests covering the changes;
 - [x] The doc is up-to-date;
 - [ ] *If I add a new sampler** I added/updated related [examples](https://github.com/blackjax-devs/blackjax/tree/main/examples)

Consider opening a **Draft PR** if your work is still in progress but you would like some feedback from other contributors.

### CODE REVIEW

**Comment by rlouf** [comment]
There were a few replacements left to be made in the documentation. I also rebased the commits. We need to make sure that the documentation builds correctly before merging this.

**Comment by rlouf** [comment]
It looks like the documentation was built ok. Can you do one last check before we merge this?

**Comment by junpenglao** [comment]
Went through all the examples - This is good to go.

**Review by rlouf — APPROVED** [review]
Great work! Will merge automatically when the tests pass.

**Pull request merged**
